--- a/src/mp_interface.mod.F90
+++ b/src/mp_interface.mod.F90
@@ -247,6 +247,7 @@
 
   ! alias for MPI_COMM_WORLD to be used outside this module
   INTEGER, SAVE, PUBLIC :: mp_comm_world
+  LOGICAL, SAVE, PUBLIC :: mp_comm_set = .FALSE.
   INTEGER, SAVE, PUBLIC :: mp_max_processor_name = HUGE(0)
   INTEGER, SAVE, PUBLIC :: mp_comm_null
 
@@ -301,8 +302,8 @@
        IF (ierr.NE.0) CALL stopgm('MPI_INIT','IERR.NE.0',& 
             __LINE__,__FILE__)
        ! an external interface has to set mp_comm_world
-       mp_comm_world = MPI_COMM_WORLD
+       IF (.NOT. mp_comm_set) mp_comm_world = MPI_COMM_WORLD
     ENDIF
     CALL mpi_errhandler_set ( mp_comm_world, MPI_ERRORS_RETURN, ierr )
     IF (ierr/=0) CALL stopgm('MPI_INIT','mpi_errhandler_set',&
